### PR TITLE
Added safe sbuf read functions with examples

### PR DESCRIPTION
**Pull request by martinbudden**

### CODE REVIEW

**Comment by digitalentity** [comment]
I like this approach. Let's have it merged and migrate all MSP to safe reads.

**Comment by fiam** [comment]
This doesn't  compile. Funcion declarations have the Safe prefix where the call does not. The reason the tests don't fail is because HIL macro is not defined, but the code is wrong.

**Comment by fiam** [comment]
There are a few more issues. Will take a look later, but this shouldn't have been merged yet.

**Comment by digitalentity** [comment]
I'll check as well. @martinbudden please also see #2110 - this is command-specific buffer protection. Takes a little bit more flash than generic sbuf protection.

**Comment by martinbudden** [comment]
I've corrected the compile error in PR https://github.com/iNavFlight/inav/pull/2111. Funny, Travis reported the build passed.

File: src/main/common/streambuf.c [hunk 1]
```diff
@@ -107,9 +107,36 @@ void sbufReadData(const sbuf_t *src, void *data, int len)
...
+        *dst = sbufReadU32(src);
+        return true;
+    }
+    return false;
+}
+
 // reader - return bytes remaining in buffer
 // writer - return available space
-int sbufBytesRemaining(const sbuf_t *buf)
+unsigned int sbufBytesRemaining(const sbuf_t *buf)
```
Comment (digitalentity) [inline src/main/common/streambuf.c:139]: Using `unsigned` here is probably a bad idea. This way if buffer overflow already happened for some reason all "Safe" functions will happily succeed overwriting even more data in process.

**Pull request merged**